docs/proposal/20241003-data-disk-support.md

```diff
@@ -0,0 +1,170 @@
+# CAPV Additional Data Disks When Creating New Machines
```

Comment:
a bit surprising to have the proposal document as part of the implementation PR.
this sort of implies "here is the implementation, and there is a proposal for it too".

Reply:
Sorry, I wasn't sure the exact process for this.  I can remove this and do separately or just remove entirely.

Reply:
np. yeah, normally the feature author sends the PR with the proposal and maintainers rally on it with comments.
then the implementation PR follows after the design is polished.

but you can move it as the first commit, i guess.

Reply:
Sounds good.  Thanks!

Reply:
I switch commit order